src/core/cgroup.c

```diff
@@ -774,13 +774,58 @@ void cgroup_oomd_xattr_apply(Unit *u, const char *cgroup_path) {
                 unit_remove_xattr_graceful(u, cgroup_path, "user.oomd_omit");
 }
 
+int cgroup_log_xattr_apply(Unit *u, const char *cgroup_path) {
+        ExecContext *c;
+        char *pattern;
+        /* Use _cleanup_free_ instead of _cleanup_strv_free_ because we need to clean the strv only, not
+         * the strings owned by ExecContext. */
+        _cleanup_free_ char **patterns_strv = NULL;
+        _cleanup_free_ char *patterns_joined = NULL;
+        size_t len;
+        int r;
+
+        assert(u);
+
+        c = unit_get_exec_context(u);
+        if (!c)
+                return log_debug_errno(-EINVAL, "Failed to get ExecContext from Unit: %m");
```

Comment:
I'd silently return in this case. both scope and slice units are unit type where we have a cgroup context but no exec context, and I think it's wise to just make things NOP for those.

hence, please just do a silent "return 0" if the exec context can't be acquired